src/index.js

```diff
@@ -37,19 +38,35 @@ function getAlgoliaSearch(params) {
   return algoliasearch(apiKey, appID);
 }
 
-function getAffectedPaths(params /* , config */) {
-  if (params.mountpoint) {
-    // TODO: translate external paths and check domain in config
-  }
-  const {
-    path: singlePath,
-    paths: multiplePaths,
-  } = params;
+/**
+ * Return source hash for an item.
+ */
+function getSourceHash(driveId, itemId) {
+  const sourceLocation = `/drives/${driveId}/items/${itemId}`;
+  return crypto.createHash('sha1').update(sourceLocation).digest('base64').substring(0, 16);
+}
 
-  if (!(singlePath || multiplePaths)) {
-    throw new Error('path/paths parameter missing.');
+/**
+ * Return items to be indexed in a common format for paths given as
+ * a string or string array or payloads in common format sent by
+ * external listeners.
+ *
+ * @param {object} params parameters to action
+ */
+function getItemsCollection(params) {
+  if (params.changes) {
+    const { driveId } = params.provider;
```

Comment:
what if it's not a onedrive change?

Reply:
Again, do we have a format definition for external change listeners that do not originate in `onedrive` ?